### PR TITLE
Add a new blog

### PR DESCRIPTION
**Pull request by husiyu**
@AndreMouche PTAL

### CODE REVIEW

**Comment by husiyu** [comment]
@MyonKeminta PTAL

File: best-practice-optimistic-transaction.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+* 事务 A、事务 B 会同时去更新同一行数据。
+* 时间点 `t4` 时，事务 A 想要更新 `id = 1` 的这一行数据，虽然此时这行数据在 `t3` 这个时间点被事务 B 已经更新了，但是因为 TiDB 乐观事务只有在事务 commit 时才检测冲突，所以时间点 `t4` 的执行成功了。
+* 时间点 `t5`，事务 B 成功提交，数据落盘。
+* 时间点 `t6`，事务 A 尝试提交，检测冲突时发现 `t1` 之后有新的数据写入，返回冲突，事务 A 提交失败，提示客户端进行重试。
+
+根据乐观锁的定义，这样做完全符合逻辑。
+
+### 重试机制
+
+我们知道了乐观锁下事务的默认行为，可以知道在冲突比较大的时候，Commit 很容易出现失败。然而，TiDB 的大部分用户，都是来自于 MySQL而 MySQL 内部使用的是悲观锁，对应到这个 case，就是事务 A 在 `t4` 更新时就会报失败，客户端就会根据需求去重试。
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]:
```suggestion
我们知道了乐观锁下事务的默认行为，可以知道在冲突比较大的时候，Commit 很容易出现失败。然而，TiDB 的大部分用户，都是来自于 MySQL；而 MySQL 内部使用的是悲观锁。对应到这个 case，就是事务 A 在 `t4` 更新时就会报失败，客户端就会根据需求去重试。
```

File: best-practice-optimistic-transaction.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+* [`tidb_disable_txn_auto_retry`](https://pingcap.com/docs-cn/v3.0/reference/configuration/tidb-server/tidb-specific-variables/#tidb_disable_txn_auto_retry)：这个参数控制是否自动重试，默认为 `1`，即不重试。
+
+* [`tidb_retry_limit`](https://pingcap.com/docs-cn/v3.0/reference/configuration/tidb-server/tidb-specific-variables/#tidb_retry_limit)：用来控制重试次数，注意只有第一个参数启用时该参数才会生效。
+
+如何设置以上参数呢？推荐两种方式设置：
+
+1. session 级别设置：
+
+	```
+	set  @@tidb_disable_txn_auto_retry = 0;
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]:
```suggestion
	set @@tidb_disable_txn_auto_retry = 0;
```

File: best-practice-optimistic-transaction.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+	set @@tidb_retry_limit=10;
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]:
```suggestion
	set @@tidb_retry_limit = 10;
```

File: best-practice-optimistic-transaction.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+	```
+
+2. 全局设置：
+
+	```
+	set  @@global.tidb_disable_txn_auto_retry = 0;
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]:
```suggestion
	set @@global.tidb_disable_txn_auto_retry = 0;
```

File: best-practice-optimistic-transaction.md [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+
+在真实使用时，如果业务场景能够预判断写入不存在冲突，如导入数据操作，建议关闭。
+
+相应地，TiKV 内存中的冲突检测也有一套类似的东西。不同的是，TiKV 的检测会更严格，不允许关闭，只提供了一个 hash 取模值的配置项：
+
+```
+scheduler-concurrency
+	* scheduler 内置一个内存锁机制，防止同时对一个 key 进行操作。每个 key hash 到不同的槽。
+	* 默认值：2048000
+```
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]: Ditto

File: best-practice-optimistic-transaction.md [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+	
+	f. TiDB 向 Primary Key 所在 TiKV 发起第二阶段提交 commit 操作，TiKV 收到 commit 操作后，检查数据合法性，清理 prewrite 阶段留下的锁。
+	
+	g. TiDB 收到 f 成功信息。
+
+6. TiDB 向客户端返回事务提交成功。
+
+7. TiDB 异步清理本次事务遗留的锁信息。
+
+### 优缺点分析
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]: This is `##` in the original draft. I think both `##` and `###` is ok though.

File: best-practice-optimistic-transaction.md [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+为了降低网络交互对于小事务的影响，我们建议小事务打包来做。如在 auto commit 模式下，下面每条语句成为了一个事务：
+
+```
+# original version with auto_commit
+UPDATE my_table SET a='new_value' WHERE id = 1; 
+UPDATE my_table SET a='newer_value' WHERE id = 2;
+UPDATE my_table SET a='newest_value' WHERE id = 3;
+```
+
+以上每一条语句，都需要经过两阶段提交，网络交互就直接 *3， 如果我们能够打包成一个事务提交，性能上会有一个显著的提升，如下：
```
Comment (MyonKeminta) [inline best-practice-optimistic-transaction.md]:
```suggestion
以上每一条语句，都需要经过两阶段提交，网络交互就直接 \*3， 如果我们能够打包成一个事务提交，性能上会有一个显著的提升，如下：
```

`*` is a meta character in markdown.

**Comment by YiniXu9506** [comment]
LGTM

**Pull request merged**